File: js/medium/TopKFrequentElements.js
class Solution {
	/**
	 * @param {number[]} nums
	 * @param {number} k
	 * @return {number[]}
	 */
	topKFrequent(nums, k) {
		const heap = new Heap((a, b) => a[1] - b[1]);
		const freq = {};
		for (const num of nums) {
			if (!freq[num]) {
				freq[num] = 0;
			}
			freq[num]++;
		}

		for (const num in freq) {
			heap.enq([parseInt(num), freq[num]]);
			if (heap.size() > k) {
				heap.deq();
			}
		}
		const result = [];
		while (heap.size() > 0) {
			result.push(parseInt(heap.deq()[0]));
		}
		return result;

	}
}

File: stonycodes70proplems/medium/TopKFrequentElements.py
from collections import Counter
import heapq 
from typing import List

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        lst,counter =[], Counter(nums)
        heapq.heapify(lst)
        for key,val in counter.items():
            heapq.heappush(lst,(val,key)) #min heap
            if len(lst)>k:
                heapq.heappop(lst)
        ans = [0]*k
        for i in range(k):
            _,num = heapq.heappop(lst)
            ans[i] = num
        return ans

